private/buf/cmd/buf/command/beta/studioagent/studioagent.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package studioagent
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/bufbuild/buf/private/buf/bufcli"
+	"github.com/bufbuild/buf/private/bufpkg/bufstudioagent"
+	"github.com/bufbuild/buf/private/pkg/app/appcmd"
+	"github.com/bufbuild/buf/private/pkg/app/appflag"
+	"github.com/bufbuild/buf/private/pkg/stringutil"
+	"github.com/bufbuild/buf/private/pkg/thread"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"golang.org/x/net/http2"
+	"golang.org/x/net/http2/h2c"
+)
+
+const (
+	// defaultShutdownTimeout is the default shutdown timeout.
+	defaultShutdownTimeout = 10 * time.Second
+	// defaultReadHeaderTimeout is the default read header timeout.
+	defaultReadHeaderTimeout = 30 * time.Second
+	// defaultIdleTimeout is the amount of time an HTTP/2 connection can be idle.
+	defaultIdleTimeout = 3 * time.Minute
+)
+
+const (
+	portFlagName              = "port"
+	disallowedHeadersFlagName = "disallowed-header"
+	forwardHeadersFlagName    = "forward-header"
+	caCertFlagName            = "ca-cert"
+	clientCertFlagName        = "client-cert"
+	clientKeyFlagName         = "client-key"
+	serverCertFlagName        = "server-cert"
+	serverKeyFlagName         = "server-key"
+)
+
+// NewCommand returns a new Command.
+func NewCommand(
+	name string,
+	builder appflag.Builder,
+) *appcmd.Command {
+	flags := newFlags()
+	return &appcmd.Command{
+		Use:   name + " <origin>",
+		Short: "Run an HTTP(S) server as the studio agent with the origin be set as the allowed origin for CORS options.",
```

Comment:
I think we should capitalise `Studio` (both here and in our documentation).

nit: "Run an HTTP(S) server as the Studio agent with the origin set as the allowed origin for CORS options."